lib/producer/events_producer.ex

```diff
@@ -41,19 +41,20 @@ defmodule Mississippi.Producer.EventsProducer do
     queue_count = init_opts[:total_count]
     queue_prefix = init_opts[:prefix]
 
-    case init_producer(events_exchange_name) do
-      {:ok, channel} ->
-        {:ok,
-         %State{
-           channel: channel,
-           events_exchange_name: events_exchange_name,
-           queue_total_count: queue_count,
-           queue_prefix: queue_prefix
-         }}
-
-      {:error, reason} ->
-        {:stop, reason}
-    end
+    state = %State{
+      channel: nil,
+      events_exchange_name: events_exchange_name,
+      queue_total_count: queue_count,
+      queue_prefix: queue_prefix
+    }
+
+    {:ok, init_producer(state)}
+  end
+
+  @impl true
+  def handle_call({:publish, _, _}, _from, %State{channel: nil} = state) do
+    # We're currently in reconnecting state
+    {:reply, {:error, :closing}, state}
```

Comment:
`:closing` came from the AMQP client library, and the channel is not being closed right now. I would suggest a more specific reason, like
```suggestion
    {:reply, {:error, :reconnecting}, state}
```

Reply:
AMQP erorrs are defined as [blocked or closing in the AMQP library](https://hexdocs.pm/amqp/AMQP.Basic.html#t:error/0), which is also the expected value from the [typespec](https://github.com/secomind/mississippi/blob/e17b5866daad54d07479c3d0b6f853aa898c58ba/lib/producer/events_producer.ex#L30). Should we instead change it to also return custom errors?

Reply:
I'm in for it, as this would not be an AMQP error but a Mississippi error.
The fact that AMQP is the transport is - or better, will be - an implementation detail